Add mention of select component's empty selection item and Java examples (#653)
vaadin/docs: articles/ds/components/select/index.asciidoc

----
--

== Empty Selection Item (Flow Only)

An empty item can be set as the first option.

[.example]
--

[source,java]
----
include::{root}/src/main/java/com/vaadin/demo/component/select/SelectEmptySelection.java[render,tags=snippet,indent=0,group=Java]
----
--

=== Customizing Empty Selection Caption

> The label for the empty item can be customized.

Reviewer: ```suggestion
The label for the empty item is customizable.
```
Tried to remove usage of "can be".
